ml-testing-toolkit-cli/templates/job.yaml

```
@@ -14,19 +14,33 @@ spec:
       backoffLimit: {{ .Values.backoffLimit }}
       restartPolicy: {{ .Values.restartPolicy }}
       {{- include "ml-testing-toolkit-cli.jobtemplate" . | nindent 6 }}
-{{- else if not .Values.tests.enabled }}
+{{- end }}
+{{- if .Values.job.enabled }} ## added job definition to values.yaml
 apiVersion: {{ template "ml-testing-toolkit-cli.apiVersion.Job" . }}
 kind: Job
 metadata:
+  {{- if .Values.job.generateNameEnabled }} ## add support for random name generation, value: true/false
+  generateName: {{ template "ml-testing-toolkit-cli.fullname" . }}-
+  {{- else }}
   name: {{ template "ml-testing-toolkit-cli.fullname" . }}
-  {{- if .Values.postInstallHook.enabled }}
-  annotations:
+  {{- end }}
+  annotations: 
+   ## If the postInstallHook is enabled without custom annotations use the helm default hook 
+  {{- if ( and .Values.postInstallHook.enabled (not .Values.job.annotations)) }}
```

Why do we need the `(not .Values.job.annotations)` portion here?

I think we should set the annotations if `postInstallHook` is set, as that is a Helm specific thing.

```suggestion
  {{- if ( .Values.postInstallHook.enabled) }}
```

It does not stop one from setting additional "custom" annotations on line [#35](https://github.com/mojaloop/helm/pull/580/files#diff-0a645a8494e1a136e062f128763627ff0dc555bd06e3db5d6d716bd6b60cdb53R35) if required.

Otherwise we should rather move those helm specific annotations as "default" values for the annotations.

Let me know what you think?